javascripts/discourse/api-initializers/discourse-mermaid-theme-component.js

```diff
@@ -58,9 +59,55 @@ async function applyMermaid(element, key = "composer") {
       mermaid.dataset.processed = true;
       mermaid.querySelector(".spinner")?.remove();
     }
+
+    if (key === "composer") {
+      debounce(() => updateMarkdownHeight(mermaid, index), 500);
+    }
   });
 }
 
+function updateMarkdownHeight(mermaid, index) {
+  let height = parseInt(mermaid.getBoundingClientRect().height);
+  let calculatedHeight = parseInt(mermaid.dataset.calculatedHeight);
+
+  if (height === 0) {
+    return;
+  }
+
+  if (height !== calculatedHeight) {
+    mermaid.dataset.calculatedHeight = height;
+    // TODO: need to use API here
+    let composer = document.getElementsByClassName("d-editor-input")[0];
```

Comment:
Can just do `document.querySelector(".d-editor-input")`, that will return the first matching element of that class

Reply:
oh ... I mean some sort of API to hunt for the composer, vs getElementsByClassName